### ISSUE TITLE
The main part of a backup flow is missing at Keycard migration flow and Create User flow

### ISSUE THREAD

**Issue by mariia-skrypnyk** [status-im/status-react#21652]
# Bug Report
https://github.com/status-im/status-mobile/pull/21467
https://github.com/status-im/status-mobile/pull/21623

## For developer:
**It's already implemented in the wallet when creating new keypair, UI should be reused** 

## Problem 
[comment]: # (A problem description)

We only have two first screens of flow and then backup is successfully done.
A big part of flow is missing.

### Reproduction

A) Migration

1. Install the app
2. Create a user
3. Go to Profile -> Keycard
4. Start the migration flow till you see screen with a backup

B) Create user 

1. Install the app
2. Create a user with empty Keycard

#### Actual behavior

https://github.com/user-attachments/assets/0b15e53b-b59c-4cde-ab15-620ccd216406

#### Expected behavior

https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=3369-58996&node-type=frame&t=4Osm8WMs7FB6Q0iU-0

![Image](https://github.com/user-attachments/assets/eeb638e6-c782-41e1-bc48-aadd6324769d)

### Additional Information

- Status version: [nightly 20](https://github.com/status-im/status-mobile/pull/21467)
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

**Comment by ilmotta** [comment]
According to analysis from QA, a user who just migrated to a profile key pair to a Keycard can still backup the seed phrase from `Settings`, as usual for normal profiles. The backup flows in Settings use the old UI components, but we think it's acceptable for 2.32 and we really don't have capacity to finish this issue in 2.32.

@churik @guylouis

**Comment by churik** [comment]
Ok, so, it is not like that @ilmotta 

1) If your phrase is not backed up before migration, you're backing up **during migration** and after that backup seed phrase is not available in profile, which seems to be expected, as you already back it up. 

2) After account creation you can see the phrase when you tap on "Backup seed phrase", so the user can see the seed phrase during onboarding and "Backup seed phrase" is not available in Profiles

**In both of these cases the seed phrase is shown to users, but entering 2 random words from it part is missing**

THe bottom line: user can see the seed phrase and enabling all checkmarks, but we do not verify that it is really backed up.

Hopefully this info will help to identify priority.

**Comment by mariia-skrypnyk** [comment]
Hi @ilmotta !

I've added that issue exist not only on Migration but on Create user.
Let me know if you need any help.

**Comment by ilmotta** [comment]
> In both of these cases the seed phrase is shown to users, but entering 2 random words from it part is missing

Thanks @mariia-skrypnyk @churik. Unfortunately my assessment of keycard flows is limited until I go back home because I can't reset my keycard, so I can end up making these silly judgmental mistakes. Thanks for all the explanation.

So coming back to the quote above:

The user can see the seed phrase before committing to the profile creation or migration for a keycard. Also, the user must press on `I have written it down on paper` before proceeding. In Metamask it's a bit different and I would be more concerned if our implementation was identical to Metamask, not because Metamask is doing anything wrong, but in the step where the pass phrase is revealed, they only show a button saying "Next". In Status 2.32, because we reveal all the words and show the button `I have written it down on paper` we have a "final" user consent that we can proceed with profile creation/migration assuming the user backed up the seed phrase.

I believe implementing the [backup flow according to Figma](https://www.figma.com/design/YGm3igIOAcwMqUVJWCJ6f1/Keycard?node-id=3395-33042&node-type=instance&m=dev) would take a significant amount of time, including testing and design review, and it's time we don't have. Therefore this issue is not a release blocker and I prefer if we push it to 2.33 to leave room for the team to handle other priorities (new stuff might appear).

cc @flexsurfer @xAlisher @guylouis

Does that plan sound reasonable to y'all? If not, please do share why.

-----

More on the opinion side, but I much prefer Metamask's seed phrase backup verification on mobile because the user must verify all words. Our current solution and the new one in Figma only verifies a subset of the words, which means the user can easily miss one or more words, get confused by the order, typos, etc. If the goal of the backup verification is to help users truly save the correct words I think Metamask's implementation is superior.

**Comment by guylouis** [comment]
the most important thing is to not break the promise of a hardware wallet: if the status profile is protected by a keycard (whatever the way the user has set up its profile) then the seed should not be stored anymore in the app
